File: sinatra/api/vrp_api.rb
require 'json'


MyApp.add_route('POST', '/api/1/vrp/optimize', {
  "resourcePath" => "/Vrp",
  "summary" => "Solves vehicle routing problems",
  "nickname" => "post_vrp", 
  "responseClass" => "JobId", 
  "endpoint" => "/vrp/optimize", 
  "notes" => "This endpoint for solving vehicle routing problems, i.e. traveling salesman or vehicle routing problems, and returns the solution.",
  "parameters" => [
    {
      "name" => "key",
      "description" => "your API key",
      "dataType" => "string",
      "paramType" => "query",
      
      "allowableValues" => "",
      
    },
    {
      "name" => "body",
      "description" => "Request object that contains the problem to be solved",
      "dataType" => "Request",
      "paramType" => "body",
    }
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

File: dart/lib/api/vrp_api.dart
part of swagger.api;



class VrpApi {
  final ApiClient apiClient;

  VrpApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Solves vehicle routing problems
  ///
  /// This endpoint for solving vehicle routing problems, i.e. traveling salesman or vehicle routing problems, and returns the solution.
  Future<JobId> postVrp(String key, Request body) async {
    Object postBody = body;

    // verify required params are set
    if(key == null) {
     throw new ApiException(400, "Missing required param: key");
    }
    if(body == null) {
     throw new ApiException(400, "Missing required param: body");
    }

    // create path and map variables
    String path = "/vrp/optimize".replaceAll("{format}","json");

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
      queryParams.addAll(_convertParametersForCollectionFormat("", "key", key));
    
    List<String> contentTypes = ["application/json"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'POST',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'JobId') as JobId ;
    } else {
      return null;
    }
  }
}
